src/main/java/com/codingmonkey/studentmanagement/controller/TeacherRestController.java

```diff
@@ -54,18 +55,23 @@ public ResponseEntity<Map<String, List<TeacherDTO>>> getTeacher(@RequestParam(va
       LOGGER.info("Get [{}] [{}] teacher details call received", firstName, lastName);
       teachers = teacherService.getTeacherByFirstNameAndLastName(firstName, lastName);
     }
-    Map<String, List<TeacherDTO>> response = Map.of("teachers", teachers);
+    Map<String, List<TeacherResponseDTO>> response = Map.of("teachers", teachers);
     return ResponseEntity.status(HttpStatus.OK).body(response);
   }
 
-  @PostMapping(consumes = APPLICATION_JSON_VALUE)
-  public ResponseEntity<TeacherDTO> addTeacher(@Valid @RequestBody TeacherDTO teacherDTO) {
+  @PostMapping()
+  public ResponseEntity<TeacherResponseDTO> addTeacher(@Valid @RequestBody TeacherRequestDTO teacherDTO) {
+    String logPrefix = "#addTeacherDetails(): ";
+    LOGGER.info("{} Request Received as {} ", logPrefix, teacherDTO);
     return ResponseEntity.status(HttpStatus.CREATED).body(teacherService.saveTeacherDetails(teacherDTO));
   }
 
-  @PutMapping(consumes = APPLICATION_JSON_VALUE)
-  public ResponseEntity<TeacherDTO> updateTeacher(@RequestBody TeacherDTO teacherDTO) {
-    return ResponseEntity.status(HttpStatus.CREATED).body(teacherService.saveTeacherDetails(teacherDTO));
+  @PutMapping("{teacherId}")
+  public ResponseEntity<TeacherResponseDTO> updateTeacher(@PathVariable int teacherId,
+                                                          @RequestBody TeacherRequestDTO teacherDTO) {
+    String logPrefix = "#updateTeacherDetails(): ";
+    LOGGER.info("{} Request Received as {} ", logPrefix, teacherDTO);
+    return ResponseEntity.status(HttpStatus.CREATED).body(teacherService.updateTeacherDetails(teacherId, teacherDTO));
```

Comment:
change the HTTP status to OK
```suggestion
    return ResponseEntity.status(HttpStatus.OK).body(teacherService.updateTeacherDetails(teacherId, teacherDTO));
```